src/restful/controllers/authController.js

```diff
@@ -26,69 +26,80 @@ class AuthController {
    * @returns {Object} response Object.
    */
 
-  static async signup(req, res) {
-    const { first_name, last_name, email, password, role, ...other } = req.body;
-
-    const db = admin.firestore();
-    try {
-      const user = await admin.auth().createUser({
-        email,
-        password,
-      });
-
-      const uid = user.uid;
-      await admin.auth().setCustomUserClaims(uid, { role });
-
-      const code = otpGenerator.generate(5, {
-        digits: true,
-        upperCase: false,
-        specialChars: false,
-        alphabets: false,
-      });
-
-      await db.collection("otp").doc(email).set({
-        otp: code,
-      });
-
-      const emailTemplate = sendOtpEmail({
-        name: first_name,
-        email: user.email,
-        otp: code,
-      });
-
-      const mailOptions = {
-        from: process.env.EMAIL,
-        to: user.email,
-        subject: "Creator Platform Account Verification",
-        html: emailTemplate,
-      };
-
-      const emailSent = await transporter.sendMail(mailOptions);
-
-      if (emailSent) {
-        const usersCollectionRef = db.collection("users");
-
-        await usersCollectionRef
-          .doc(user.uid)
-          .set({ uid: user.uid, first_name, last_name, role, ...other });
-
-        util.statusCode = 200;
-        util.setSuccess(200, "Success", { email, uid });
-        return util.send(res);
-      } else {
-        // Delete the user if email sending failed
-        await admin.auth().deleteUser(user.uid);
-        util.statusCode = 500;
-        util.message = "Failed to send verification email";
-        return util.send(res);
+  
+    static async signup(req, res) {
+      const { first_name, last_name, email, password, role, profile_picture, ...other } = req.body;
```

Comment:
We don't store profile picture data on sign up, but users can update their profile picture when editing their profile